modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItFloatingPointTest.java

```diff
@@ -0,0 +1,200 @@
+
+        assertEquals(1L, sql("SELECT COUNT(*) FROM test WHERE f = '+Infinity'::FLOAT").get(0).get(0));
+        assertEquals(1L, sql("SELECT COUNT(*) FROM test WHERE f = '+Infinity'::DOUBLE").get(0).get(0));
+
+        assertEquals(0L, sql("SELECT COUNT(*) FROM test WHERE f = 'NaN'::FLOAT").get(0).get(0)); // NaN never equals
+        assertEquals(0L, sql("SELECT COUNT(*) FROM test WHERE f = 'NaN'::DOUBLE").get(0).get(0)); // NaN never equals
+
+        assertEquals(2L, sql("SELECT COUNT(*) FROM test WHERE f > 0").get(0).get(0)); // 1.0, +Infinity
+        assertEquals(2L, sql("SELECT COUNT(*) FROM test WHERE f < 0").get(0).get(0)); // -Infinity, -1,0
+
+        assertEquals(5L, sql("SELECT COUNT(*) FROM test "
+                + "WHERE f IS DISTINCT FROM '+Infinity'::FLOAT AND d IS DISTINCT FROM '-Infinity'::FLOAT").get(0).get(0)); // NaN, 1.0, -1,0
+        assertEquals(5L, sql("SELECT COUNT(*) FROM test "
+                + "WHERE f IS DISTINCT FROM '+Infinity'::DOUBLE AND d IS DISTINCT FROM '-Infinity'::DOUBLE").get(0).get(0));
+
+        // NaN not equal to NaN
+        assertEquals(7L, sql("SELECT COUNT(*) FROM test WHERE f IS DISTINCT FROM 'NaN'::FLOAT").get(0).get(0));
+        assertEquals(7L, sql("SELECT COUNT(*) FROM test WHERE d IS DISTINCT FROM 'NaN'::DOUBLE").get(0).get(0));
+    }
+
+    @Test
+    void testAggregations() {
+        List<List<Object>> minRows = sql("SELECT MIN(f), MIN(d) FROM test");
+        List<List<Object>> maxRows = sql("SELECT MAX(f), MAX(d) FROM test");
+        List<List<Object>> avgRows = sql("SELECT AVG(f), AVG(d) FROM test");
+
+        assertThat(minRows.get(0), contains(Float.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY));
+        assertThat(maxRows.get(0), contains(Float.NaN, Double.NaN));
+        assertThat(avgRows.get(0), contains(Double.NaN, Double.NaN));
+    }
+
+    @Test
+    void testGrouping() {
+        // Insert more data.
+        sql("INSERT INTO test VALUES (?, ?, ?)", 8, Float.POSITIVE_INFINITY, Double.POSITIVE_INFINITY);
+        sql("INSERT INTO test VALUES (?, ?, ?)", 9, Float.NEGATIVE_INFINITY, Double.NEGATIVE_INFINITY);
+        sql("INSERT INTO test VALUES (?, ?, ?)", 10, Float.NaN, Double.NaN);
+        sql("INSERT INTO test VALUES (?, ?, ?)", 11, -1.0f, -1.0d);
+        sql("INSERT INTO test VALUES (?, ?, ?)", 12, 1.0f, 1.0d);
+        sql("INSERT INTO test VALUES (?, ?, ?)", 13, -0.0f, -0.0d);
+        sql("INSERT INTO test VALUES (?, ?, ?)", 14, +0.0f, +0.0d);
+
+        assertThat(sql("SELECT * FROM test"), hasSize(14));
+
+        List<Float> floats = sql("SELECT f FROM test GROUP BY f")
+                .stream().flatMap(List::stream).map(Float.class::cast).collect(toList());
+        List<Double> doubles = sql("SELECT d FROM test GROUP BY d")
+                .stream().flatMap(List::stream).map(Double.class::cast).collect(toList());
+
+        assertThat(floats, containsInAnyOrder(Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY, Float.NaN, -1.0f, 1.0f, 0.0f, -0.0f));
```

Comment:
is it ok - to await any ordering here ? I think it need to be deterministic, did it ?

Reply:
Why do expect any ordering here?
I expect all duplicated values are merged into group. The ordering make no sense here.